routes/clams.router.js

```diff
@@ -0,0 +1,160 @@
+const toUnnamed = require('named-placeholders')();
+const { pool } = require('../server/db');
+const { keysToCamel, isNumeric } = require('../common/utils');
+
+const router = express.Router();
+
+// Get data for all clams
+router.get('/', async (req, res) => {
+  try {
+    const clams = await pool.query('SELECT * FROM clam;');
+    res.status(200).json(keysToCamel(clams));
+  } catch (err) {
+    res.status(400).send(err.message);
+  }
+});
+
+// Get data for specific clam based on clamId
+router.get('/:clamId', async (req, res) => {
+  try {
+    const { clamId } = req.params;
+    isNumeric(clamId);
+
+    const [query, params] = toUnnamed('SELECT * FROM clam WHERE clamId = :clamId', { clamId });
+    const clam = await pool.query(query, params);
+
+    res.status(200).json(keysToCamel(clam));
+  } catch (err) {
+    res.status(400).send(err.message);
+  }
+});
+
+// Get data for clams based on rakerId
+router.get('/rakers/:rakerId', async (req, res) => {
+  try {
+    const { rakerId } = req.params;
+    isNumeric(rakerId);
+
+    const [query, params] = toUnnamed('SELECT * FROM clam WHERE rakerId = :rakerId', { rakerId });
+    const clams = await pool.query(query, params);
+
+    res.status(200).json(keysToCamel(clams));
+  } catch (err) {
+    res.status(400).send(err.message);
+  }
+});
+
+// Create clam
+router.post('/', async (req, res) => {
+  try {
+    const { clamId, rakerId, lat, lon, length, width, weight, comments, image } = req.body;
+
+    isNumeric(clamId);
+    isNumeric(rakerId);
+
+    const [query, params] = toUnnamed(
+      `
+      INSERT INTO \`clam\` (
+        clamId,
+        rakerId,
+        lat,
+        lon,
+        length,
+        width,
+        weight,
+        comments,
+        image
+        )
+      VALUES (
+        :clamId,
+        :rakerId,
+        :lat,
+        :lon,
+        :length,
+        :width,
+        :weight,
+        :comments,
+        :image
+      );
+      SELECT * FROM clam WHERE clamId = :clamId`,
+      {
+        clamId,
+        rakerId,
+        lat,
+        lon,
+        length,
+        width,
+        weight,
+        comments,
+        image,
+      },
+    );
+
+    const clam = await pool.query(query, params);
+
+    res.status(200).json(keysToCamel(clam));
+  } catch (err) {
+    res.status(400).send(err.message);
+  }
+});
+
+// Delete clam
+router.delete('/:clamId', async (req, res) => {
+  try {
+    const { clamId } = req.params;
+    isNumeric(clamId);
+
+    const [query, params] = toUnnamed('DELETE FROM clam WHERE clamId = :clamId', { clamId });
+    await pool.query(query, params);
+
+    res.status(200).json(keysToCamel(`Deleted clam #${clamId}`));
+  } catch (err) {
+    res.status(400).send(err.message);
+  }
+});
+
+// Update clam
+router.put('/:clamId', async (req, res) => {
+  try {
+    const { clamId } = req.params;
+    isNumeric(clamId);
+
+    const { rakerId, lat, lon, length, width, weight, comments, image } = req.body;
```

Comment:
`lat`, `lon`, `length`, `width`, `weight` can be checked with `isNumeric` too here since they are all numbers

edit: on the `POST` route too